enterprise/reporting/error_extractor.go

```diff
@@ -345,32 +353,18 @@ func getErrorCodeFromStatTags(statTags map[string]string) string {
 	return strings.Join(errorCodeParts, ":")
 }
 
-func (ext *ExtractorHandle) isVersionDeprecationError(errorMessage string) bool {
-	// Convert to lowercase for case-insensitive matching
-	cleanedError := strings.Replace(strings.ToLower(errorMessage), "-", " ", -1)
-
-	// Check for version-status combinations
-	for _, vPattern := range versionPatterns {
-		if !strings.Contains(cleanedError, vPattern) {
-			continue
-		}
-
-		for _, sPattern := range statusPatterns {
-			if strings.Contains(cleanedError, sPattern) {
-				// Found a version pattern and a status pattern
-				return true
-			}
-		}
-	}
-
-	// Check for strong deprecation terms (using pre-computed list)
-	for _, term := range strongDeprecationTerms {
-		if strings.Contains(cleanedError, term) {
-			return true
-		}
-	}
+func containsDeprecationKey(errorMessage, key string) bool {
+	return strings.HasPrefix(errorMessage, key) || strings.Contains(errorMessage, " "+key)
+}
 
-	return false
+func (ext *ExtractorHandle) isVersionDeprecationError(errorMessage string) bool {
+	// Normalize error message
+	cleanedError := strings.ReplaceAll(strings.ToLower(errorMessage), "-", " ")
+	return slices.ContainsFunc(deprecationKeywordSets, func(keywordSet []string) bool {
```

Comment:
keywords like version, api, endpoint are repeated and if I'm not wrong, we're doing string comparison multiple times. Can this be improved by doing this in two passes? one for repeated keywords and then unique keywords?

Reply:
updated